packages/dropdowns/src/Menu/Menu.js

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, { createContext } from 'react';
+import PropTypes from 'prop-types';
+import { Popper } from 'react-popper';
+import { withTheme, isRtl } from '@zendeskgarden/react-theming';
+
+import { StyledMenu } from '../styled/index.js';
+import useDropdownContext from '../utils/useDropdownContext';
+import { getPopperPlacement, getRtlPopperPlacement } from '../utils/garden-placements';
+
+export const MenuContext = createContext();
+
+/**
+ * Accepts all `<ul>` props
+ */
+const Menu = props => {
+  const {
+    placement,
+    popperModifiers,
+    eventsEnabled,
+    animate,
+    maxHeight,
+    style: menuStyle,
+    zIndex,
+    ...otherProps
+  } = props;
+  const {
+    itemIndexRef,
+    previousItemRef,
+    previousIndexRef,
+    nextItemsHashRef,
+    popperReferenceElementRef,
+    downshift: { isOpen, getMenuProps }
+  } = useDropdownContext();
+
+  // Reset Downshift refs on every render
+  itemIndexRef.current = 0;
+  nextItemsHashRef.current = {};
+  previousItemRef.current = undefined;
+  previousIndexRef.current = undefined;
```

Comment:
How come you need to do this?

Reply:
Since the Downshift action `selectItemAtIndex` is index based, we keep the ref of the most recent (should only ever be 1) `PreviousItem`.

The `previousItemRef` is actually left over from a previous implementation. I've now removed it.